aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -278,6 +288,11 @@ func resourceAwsLambdaEventSourceMappingUpdate(d *schema.ResourceData, meta inte
 		Enabled:      aws.Bool(d.Get("enabled").(bool)),
 	}
 
+	// AWS API will fail if this parameter is set (even as default value) for sqs event source.  Ideally this should be implemented in GO SDK or AWS API itself.
+	if !strings.HasPrefix(d.Get("event_source_arn").(string), "arn:aws:sqs:") {
```

Comment:
Can you please update this check to use [`arn.Parse()`](https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/#Parse)? There are additional AWS partitions outside Standard, such as GovCloud (US) and China. e.g. with a new `"github.com/aws/aws-sdk-go/aws/arn"` import

```suggestion
	eventSourceArn, err := arn.Parse(d.Get("event_source_arn").(string))

	if err == nil && eventSourceArn.Service != "sqs" {
```

Reply:
👍